lib/src/tree_controller.dart

```diff
@@ -197,6 +197,26 @@ class TreeController<T extends Object> with ChangeNotifier {
     rebuild();
   }
 
+  /// Expands all nodes of this tree consequently.
+  void expandAll() => expandCascading(roots);
+
+  /// Collapses all nodes of this tree consequently.
+  void collapseAll() => collapseCascading(roots);
+
+  /// Whether this tree is completely expanded.
+  ///
+  /// To know this it is required to know if root nodes contained in
+  /// the cache of expanded nodes.
+  bool get isTreeExpanded =>
+      roots.every((node) => _expandedNodes.contains(node));
+
+  /// Whether this tree is completely collapsed.
+  ///
+  /// To know this it is required to know if root nodes does not contained in
+  /// the cache of expanded nodes.
+  bool get isTreeCollapsed =>
+      roots.every((node) => !_expandedNodes.contains(node));
```

Comment:
> Since both getters are not traversing the tree, a better name might be allRootsExpanded and allRootsCollapsed.

So then `areRootsExpanded` and `areRootCollapsed` because of booleans. But in regards of quantity of roots user see the **tree** and he does not care what is underhood. It is important to know if the whole tree expanded or collapsed. I think my names more intuitive and user-friendly.

> About the implementation, please default to using getExpansionState instead of _expandedNodes.contains() 

Yes. I agree.

Reply:
If I stumbled across a boolean called `isTreeExpanded` I would assume every single node in every single level is expanded. For the `isTreeCollapsed` I think it is not an issue, but I would rename it anyway for consistency.

Reply:
Maybe let's settle with `areAllRootsExpanded` and `areAllRootsCollapsed`?

Reply:
What if we live `isTreeExpanded` and `isTreeCollapsed` but use such code:
```dart
  bool get isTreeExpanded {
    final totalNodes = [], expandedNodes = [];
    depthFirstTraversal(
      onTraverse: (entry) {
        totalNodes.add(entry);
        if (entry.isExpanded) {
          expandedNodes.add(entry);
        }
      },
      descendCondition: (node) => true,
    );
    return totalNodes.length == expandedNodes.length;
  }
```
```dart
  bool get isTreeCollapsed {
    final totalNodes = [], collapsedNodes = [];
    depthFirstTraversal(
      onTraverse: (entry) {
        totalNodes.add(entry);
        if (!entry.isExpanded) {
          collapsedNodes.add(entry);
        }
      },
      descendCondition: (node) => true,
    );
    return totalNodes.length == collapsedNodes.length;
  }
```

Reply:
Sure, thats great. To avoid wasting more of your time, we can merge this PR with the suggested changes above and I'll do a followup PR later today to apply some performance improvements I thought of, what do you think?